### PR TITLE
[FEATURE] Add HTTP endpoint to manage global variable

### PR DESCRIPTION
**Pull request by Nexucis**
This PR is adding two new resources in the backend : `GlobalVariable` and `ProjectVariable`. These variables can be then shared and used across different dashboard.

This is something coming up in the discussion https://github.com/perses/perses/discussions/226

For example:

```bash
curl -XGET https://perses.dev/api/v1/globalvariables
```
```json
[
{
  "kind": "GlobalVariable",
  "metadata": {
    "name": "MyList"
  },
  "spec": {
    "kind": "ListVariable",
    "spec": {
      "display": {
        "name": "my awesome variable",
        "hidden": false
      },
      "default_value": [
        "default1",
        "default2"
      ],
      "allow_all_value": false,
      "allow_multiple": true,
      "plugin": {
        "kind": "PrometheusLabelValuesVariable",
        "spec": {
          "label_name": "instance",
          "matchers": [
            "up"
          ]
        }
      }
    }
  }
}
]
```

### CODE REVIEW

**Comment by sjcobb** [comment]
It'd be cool someday to have an OpenAPI spec that is always update to with the various REST endpoints in Perses (not sure best lib to use, this one seems popular? https://github.com/go-swagger/go-swagger)

Or even something like a Postman collection that we can commit and share (useful, but hard to keep up to date)

**Comment by Nexucis** [comment]
> It'd be cool someday to have an OpenAPI spec that is always update to with the various REST endpoints in Perses (not sure best lib to use, this one seems popular? https://github.com/go-swagger/go-swagger)
> 
> Or even something like a Postman collection that we can commit and share (useful, but hard to keep up to date)

On the paper it can be a good idea. But in reality, the issue it will result is that the swagger generated from the go code won't contain the full definition of the datamodel (since some piece of it is in cuelang)

So the swagger can provide you the full list of endpoint but with incomplete datamodel which can result of a wrong usage of the API.
So I would be quite cautious to add that.

But if have somewhere the complete data model for the dashboard maybe we could like it in the swagger and that could solve this issue.
So I think the priority would be more to display the complete datamodel in a nice way (maybe interactive way)

File: pkg/model/api/v1/variable.go [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+
+func (v *GlobalVariable) GetKind() string {
+	return string(v.Kind)
+}
+
+func (v *GlobalVariable) GetSpec() interface{} {
+	return v.Spec
+}
+
+type Variable struct {
```
Comment (Gladorme) [inline pkg/model/api/v1/variable.go:87]: IMO, I would prefer naming `Variable` => `ProjectVariable` even if it's longer, in order to not think it's a dashboard variable 😄

Comment (Nexucis) [reply pkg/model/api/v1/variable.go:87]: in a REST API terms, having a path starting by `/projects/perses/<resources>` already mean the resources is part of a project. So naming a resource `project<resource>` is redundant :/

**Comment by Nexucis** [comment]
> It'd be cool someday to have an OpenAPI spec that is always update to with the various REST endpoints in Perses (not sure best lib to use, this one seems popular? https://github.com/go-swagger/go-swagger)
> 
> Or even something like a Postman collection that we can commit and share (useful, but hard to keep up to date)

apparently @celian-garcia is trying something internally to generate the swagger in a nice way and it should give some nice result. So probably we will be able to have it here as well :)

**Comment by Nexucis** [comment]
but the datamodel will still be incomplete until we are able to generate it in a proper documentation

File: pkg/model/api/v1/variable.go [hunk 2]
```diff
@@ -84,6 +85,8 @@ func (v *GlobalVariable) GetSpec() interface{} {
 	return v.Spec
 }
 
+// Variable is a global variable at project level.
+// If you are looking for the variable contained in a dashboard, take a look at dashboard.Variable
```
Comment (AntoineThebaud) [inline pkg/model/api/v1/variable.go:89]: Variable relates to variables defined at project level.
If you are looking for variable defined at dashboard level, see dashboard.Variable

**Comment by celian-garcia** [comment]
> It'd be cool someday to have an OpenAPI spec that is always update to with the various REST endpoints in Perses (not sure best lib to use, this one seems popular? https://github.com/go-swagger/go-swagger)
> 
> Or even something like a Postman collection that we can commit and share (useful, but hard to keep up to date)

@sjcobb @Nexucis What I'm working on in our internal tool is to move from go-swagger to swaggo. 
It looks very less verbose because go-swagger forced us to define some structures for the requests/responses data model, and swaggo parse the code a bit better.

Here you have in addition a cuelang datamodel ? That would be complex, maybe we would have to do a cuelang > golang script to port the datamodel in something readable by swaggo or go-swagger

I can try with perses when I'm done with our internal project

**Comment by Nexucis** [comment]
I think you shouldn't take care about the partial definition you can have in cuelang.
For the moment having the swagger generated with the partial definition will already be good. Probably we will have to put some warnings in the comment on the plugin.

**Pull request merged**